### ISSUE TITLE
Bug: no-var-requires not working for calling functions

### ISSUE THREAD

**Issue by bastienmoulia** [typescript-eslint/typescript-eslint#7062]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=E4UwjgrglqAUDkBjAngB1AZwwWgLYHtEALAQwHcQN9cRtRV9gAXEYAelQBsIBzKAO3gBKWPn5CA3AFgAUEA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDcBDaWtEQBHWJSGoMkRNGgdokcGAC+IZUA

### Repro Code

```typescript
require('cypress-mochawesome-reporter/plugin')(on);
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-var-requires": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No error because there is no variables.

### Actual Result

There is an error.

**Comment by bradzacher** [comment]
This is still a require call, hence the rule reports on it. Your code should be using an import expression instead of a require.

If you fully intend to do this then use a disable comment and detail why you're working around the rule.

**Comment by bastienmoulia** [comment]
I’m not sure the require is the issue because there is no error with:
`require('cypress-mochawesome-reporter/plugin');`
